src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java

```diff
@@ -23,6 +23,33 @@
  * can appear in GATK-based applications.
  */
 public class IntervalUtils {
+
+    /**
+     * Lexicographical (contig) order comparator.
+     * <p>
+     * Intervals from different contigs order is according their enclosing
+     * contigs name ascending lexicographical order.
+     * </p>
+     * <p>
+     * Intervals from the same contigs order is according to their start
+     * position ascending numerical order, and, in case of a tie, the stop position's.
+     * </p>
+     * <p>
+     * This comparator with throw an exception with locatables with {@code null} contigs.
+     * </p>
+     */
+    public final static Comparator<Locatable> LEXICOGRAPHICAL_ORDER_COMPARATOR =
+            Comparator.comparing(Locatable::getContig,
+                    (contig1, contig2) -> {
+                        if (contig1 == null || contig2 == null) {
+                            throw new IllegalArgumentException("null contig not supported");
```

Comment:
Since unmapped records traditionally sort after mapped records, perhaps you should sort `Locatables` with null contigs after `Locatables` with non-null contigs? The `Locatable` interface explicitly allows null to be returned from `getContig()`.